x-pack/filebeat/input/awss3/s3_objects.go

```diff
@@ -293,18 +300,25 @@ func (p *s3ObjectProcessor) readFile(r io.Reader) error {
 	var offset int64
 	for {
 		message, err := reader.Next()
-		if err == io.EOF {
+
+		if len(message.Content) > 0 {
```

Comment:
This is exactly why it works, It is designed that way to handle the EOF. The EOF error is not an error per se, it is used as logic to delimit messages. In my understanding. Otherwise, the tests don't pass.

Reply:
Oh, you're right, I see. Forget about this comment! 🙈